peachjam/views/legislation.py

```diff
@@ -17,3 +22,77 @@ class LegislationListView(FilteredDocumentListView):
 class LegislationDetailView(BaseDocumentDetailView):
     model = Legislation
     template_name = "peachjam/legislation_detail.html"
+
+    def get_notices(self):
+        notices = super().get_notices()
+        repeal = self.get_repeal_info()
+        if self.object.repealed and repeal:
+            msg = "This {} was repealed on {} by <a href='{}'>{}</a>"
+            notices.append(
+                {
+                    "type": messages.ERROR,
+                    "html": format_html(
+                        msg.format(
+                            self.object.metadata_json["type_name"],
+                            repeal["date"],
+                            repeal["repealing_uri"],
+                            repeal["repealing_title"],
+                        )
+                    ),
+                }
+            )
+
+            msg = (
+                "This is the latest version of this {} as it was when it was repealed."
```

Comment:
This might not be true. If a work is repealed, then all versions of the work are marked repealed.

Instead of calculating this message here, it should be calculated when you're working out if this is the latest expression or not.

Reply:
Look at how the messages are determined for openbylaws.org.za: https://github.com/laws-africa/openbylaws.org.za/blob/a9138e619ae5d21df182ff09e621a773c0ab9d76/_includes/work_notices.html#L16